packages/react-instantsearch-hooks/package.json

```diff
@@ -42,10 +42,10 @@
     "@babel/runtime": "^7.1.2",
     "algoliasearch-helper": "^3.6.2",
     "dequal": "^2.0.0",
-    "instantsearch.js": "^4.32.0"
+    "instantsearch.js": "https://pkg.csb.dev/algolia/instantsearch.js/commit/3f954126/instantsearch.js"
   },
   "peerDependencies": {
     "algoliasearch": ">= 3.1 < 5",
-    "react": ">= 16.8.0 < 18"
+    "react": ">= 16.8.0 < 19"
```

Comment:
I tested the new APIs with React 18 and their new [SSR Stream APIs](https://github.com/reactwg/react-18/discussions/22) and they're compatible.